qcodes_contrib_drivers/drivers/Vaunix/LDA.py

```diff
@@ -0,0 +1,291 @@
+r"""
+This is the QCoDeS driver for Vaunix LDA digital attenuators. It requires the
+DLL that comes with the instrument, ``VNX_atten64.dll`` and/or
+``VNX_atten.dll``, for 64-bit Windows and 32-bit Windows, respectively. If the
+instrument has more than one physical channel, ``InstrumentChannel``s are
+created for each one. If the instrument has only one physical channel, no
+channels are created and the parameters will be assigned to this instrument
+instead. The sweep profiles available in the API are not implemented.
+
+Tested with 64-bit system and
+- LDA-133
+- LDA-802Q
+
+"""
+
+import logging
+from typing import Optional, Dict, Callable, Union, cast
+from functools import partial
+from platform import architecture
+import os
+import sys
+import ctypes
+
+from qcodes import Instrument, InstrumentChannel, Parameter
+from qcodes.utils.validators import Numbers
+
+logger = logging.getLogger(__name__)
+
+
+class Vaunix_LDA(Instrument):
+    def __init__(self, name: str,
+                 serial_number: int,
+                 dll_path: Optional[str] = None,
+                 channel_names: Optional[Dict[int, str]] = None,
+                 test_mode: Optional[bool] = False,
+                 **kwargs):
+        r"""
+        QCoDeS Instrument for Vaunix LDA digital attenuators.
+
+        Args:
+            name: Qcodes name for this instrument
+            serial_number: Serial number of the instrument, used to identify
+                it.
+            dll_path: Look for the LDA DLLs in this directory.
+                By default, the directory this file is in.
```

Comment:
i don't think this is a good default. qcodes-contrib-drivers is published on pypi and users are encouraged to install it from there (as opposed to cloning the repo) - in that case the path to this file will be buries somewhere inside virtualenv/conda environement which is not convenient, and users should not go and add files there. Hence, it's better if the default DLL path is something that the manufacturer imposes, for example, quite a few instruments that i've seen install their DLLs somewhere in program files. If the manufacturer doesn't suggest any path then simple `<user dir>` or `C:/` is much better for a default.

Reply:
Fixed in 0191808.